api/common/common.proto

```diff
@@ -5,8 +5,44 @@ package common;
 option go_package = "github.com/talos-systems/talos/pkg/machinery/api/common";
 
 import "google/protobuf/any.proto";
+import "google/protobuf/descriptor.proto";
 import "google/rpc/status.proto";
 
+// An alternative to using options could be extracting versions from comments.
+// Unfortunately, they are not available: https://github.com/golang/protobuf/issues/1134
+// Also, while option numbers can be the same,
+// names should be different: https://github.com/protocolbuffers/protobuf/issues/4861
+
+extend google.protobuf.MessageOptions {
+  // Indicates the Talos version when this deprecated message will be removed from API.
+  string remove_deprecated_message = 93117;
```

Comment:
Explaining the significance of that number is left as an exercise to the reader 😆